### DISCUSSION TITLE
[Linux]使用gcc13编译会报错

### DISCUSSION THREAD

**Discussion by wangyinlong** [Tencent/libpag#2247] [Q&A] [answered]
我在linux环境编译PAG的时候（执行./build_pag.sh ）遇到了下面报错，不知道你有没有遇到过，是不是我的c++编译环境不太对？还是其他什么问题，求指教。
使用的是gcc13.1.0
我是根据这里的指引编译的https://github.com/Tencent/libpag/blob/main/linux/README.zh_CN.md
<img width="1669" alt="" src="https://github.com/Tencent/libpag/assets/5877234/32cae863-c8b0-4759-84f0-0d78b19b5566">
<img width="839" alt="企业微信截图_78d0cb6e-6485-4a33-8c4d-0dc7bd56d297" src="https://github.com/Tencent/libpag/assets/5877234/75d4a2d3-52c2-4185-882a-2f4625ba5f5e">

所以想问下使用什么版本的gcc可以通过，12？

**Comment by domchen** [comment, marked as answer]
GCC 9.0是可以的。这个问题有时间我们也适配一下。

**Reply by domchen** [reply]
GCC 11.0也是可以的，刚刚看了github流水线使用的是这个版本。

**Reply by domchen** [reply]
开发分支main上已经增加了对gcc 13的兼容，等后续正式发版后会进入稳定分支。

**Comment by wangyinlong** [comment]
感谢
-- 

王崟垄
Yinlong Wang
School of Software
Tsinghua University
Mobile: +86-18515882258



Dom Chen ***@***.***> 于2024年4月12日周五 12:00写道：
